epochStart/bootstrap/fromLocalStorage.go

```diff
@@ -138,9 +146,10 @@ func (e *epochStartBootstrap) prepareEpochFromStorage() (Parameters, error) {
 		}
 	}
 
+	shardIDToReturn := e.applyShardIDAsObserverIfNeeded(e.shardCoordinator.SelfId())
```

Comment:
Does an Observer node have anything to do with the shardCoordinator.SelfId? If it is an observer, he is not assigned to a shard by the shardCoordinator

Reply:
he is assigned to a shard - shardCoordinator says his shard.